lib/phoenix/live_dashboard/components/tab_bar_component.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Phoenix.LiveDashboard.TabBarComponent do
+  @moduledoc false
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{page: page, entries: entries} = assigns
+    current = current_tab(page.params, entries)
+    {:ok, assign(socket, entries: entries, current: current, page: page)}
+  end
+
+  defp current_tab(%{"tab" => tab}, entries) do
+    if Enum.any?(entries, fn {id, _} -> Atom.to_string(id) == tab end) do
+      String.to_existing_atom(tab)
+    else
+      default_tab(entries)
+    end
+  end
+
+  defp current_tab(_, entries) do
+    default_tab(entries)
+  end
```

Comment:
`with` can be used to clean this up a bit:

```suggestion
  defp current_tab(params, entries) do
    with %{"tab" => tab} <- params,
         true <- Enum.any?(entries, fn {id, _} -> Atom.to_string(id) == tab end) do
      String.to_existing_atom(tab)
    else
      _ -> default_tab(entries)
    end
  end
```

WDYT?

Reply:
Nice! More elegant :)